adapters/repos/db/lsmkv/segment_bloom_filters.go

```diff
@@ -36,17 +41,32 @@ func (s *segment) bloomFilterSecondaryPath(pos int) string {
 }
 
 func (s *segment) initBloomFilters(metrics *Metrics, overwrite bool) error {
-	if err := s.initBloomFilter(overwrite); err != nil {
-		return fmt.Errorf("init bloom filter for primary index: %w", err)
-	}
+	g := enterrors.NewErrorGroupWrapper(s.logger)
+	g.SetLimit(_NUMCPU)
+	g.Go(func() error {
+		if err := s.initBloomFilter(overwrite); err != nil {
+			return fmt.Errorf("primary index: %w", err)
+		}
+		return nil
+	})
+
 	if s.secondaryIndexCount > 0 {
 		s.secondaryBloomFilters = make([]*bloom.BloomFilter, s.secondaryIndexCount)
 		for i := range s.secondaryBloomFilters {
-			if err := s.initSecondaryBloomFilter(i, overwrite); err != nil {
-				return fmt.Errorf("init bloom filter for secondary index at %d: %w", i, err)
-			}
+			i := i // capture loop var
+			g.Go(func() error {
```

Comment:
I wonder if primary index need to be created before creating the secondary index though?

Reply:
good point, my assumption at initiation level they are not dependent but lets double check